### PR TITLE
CI Improvements; more focused runs and best practices

### PR DESCRIPTION
**Pull request by pww217**
Small changes, as the title says. 

Question though: is there any overlap between the pre-commit python linting and Philipp's `lint.yaml`? The latter looks like JS linting, but want to confirm so there's no redundancy.

If they're different, may be possible to add that to `pre-commit` as well. Up to y'all.

### CODE REVIEW

**Comment by mattkram** [comment]
I've actually got a `pre-commit` hook locally which I haven't made a PR for yet which does `eslint`. I haven't spent the time to tweak settings, but we should be able to remove `lint.yml` once we get the settings right.

```yaml
-   repo: https://github.com/pre-commit/mirrors-eslint
    rev: 'v8.14.0'
    hooks:
    -   id: eslint
        additional_dependencies:
        -   eslint@>=5.16.0
        -   eslint-config-google@latest
```

**Comment by pww217** [comment]
I added that to pre-commit @mattkram but it's failing - looks like the path is wrong. Not really seeing what argument is used to change that in the docs, but it's defaulting to /code/...

**Comment by mattkram** [comment]
Oh, I may need to give you a config file. Lemme try locally.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -9,3 +9,10 @@ repos:
     hooks:
       - id: isort
         name: isort (python)
+  - repo: https://github.com/pre-commit/mirrors-eslint
+    rev: 'v8.14.0'
+    hooks:
+    -   id: eslint
+        additional_dependencies:
+        -   eslint@>=5.16.0
+        -   eslint-config-google@latest
```
Comment (cclauss) [inline .pre-commit-config.yaml:18]: ⛔️ means that GitHub wants files to end with one and only one \n

**Comment by mattkram** [comment]
@pww217 Let's keep this in `lint.yml`. Needs some more work to untangle all the dependencies.

This is tough because `eslint` needs some plugins, and also the `npm` root is not the repo root.

**Review by cclauss — APPROVED** [review]

**Comment by pww217** [comment]
Already deleted eslint from pre-commit for now.

**Pull request merged**